src/test/java/core/basesyntax/service/impl/FruitTransactionServiceImplTest.java

```diff
@@ -0,0 +1,108 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitTransactionService;
+import core.basesyntax.strategy.BalanceOperationHandler;
+import core.basesyntax.strategy.FruitTransactionStrategy;
+import core.basesyntax.strategy.FruitTransactionStrategyImpl;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.PurchaseOperationHandler;
+import core.basesyntax.strategy.ReturnOperationHandler;
+import core.basesyntax.strategy.SupplyOperationHandler;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitTransactionServiceImplTest {
+    private static FruitTransactionService fruitTransactionService;
+    private static List<FruitTransaction> transactions;
+    private static FruitTransactionStrategy strategy;
+    private static Storage fruitsStorage;
+    private static Map<FruitTransaction.Operation,OperationHandler> strategyMap;
+
+    @BeforeClass
+    public static void beforeClass() {
+        transactions = new ArrayList<>();
+        fruitsStorage = new Storage();
+        strategyMap = new HashMap<>();
+        strategy = new FruitTransactionStrategyImpl(strategyMap);
+        fruitTransactionService = new FruitTransactionServiceImpl(strategy,fruitsStorage);
+    }
+
+    @Before
+    public void setUp() {
+        strategyMap.put(FruitTransaction.Operation.BALANCE, new BalanceOperationHandler());
+        strategyMap.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseOperationHandler());
+        strategyMap.put(FruitTransaction.Operation.RETURN, new ReturnOperationHandler());
+        strategyMap.put(FruitTransaction.Operation.SUPPLY, new SupplyOperationHandler());
+    }
+
+    @Test
+    public void processTransaction_Ok() {
+        FruitTransaction fruitTransaction = new FruitTransaction();
+        fruitTransaction.setOperation(FruitTransaction.Operation.BALANCE);
+        fruitTransaction.setFruit("banana");
+        fruitTransaction.setQuantity(20);
+        transactions.add(fruitTransaction);
```

Comment:
you use the same code several times, let's eliminate duplication and create a separate method that will be creating FruitTransaction based on passed parameters